sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -325,6 +333,19 @@ public void setAttribute(String key, AttributeValue value) {
         logger.log(Level.FINE, "Calling setAttribute() on an ended Span.");
         return;
       }
+      totalAttributeCount++;
+      if (attributes.get(key) == null && attributes.size() >= maxNumberOfAttributes) {
+        logger.log(
+            Level.FINE,
+            "Span with name '"
+                + this.name
+                + "' has reached the maximum number of attributes ("
+                + maxNumberOfAttributes
+                + "). Dropping attribute with key '"
+                + key
+                + "'");
```

Comment:
I'd prefer to see the logging done with string interpolation, rather than a giant concatenated string like this.

Reply:
Done :+1: